reposerver/repository/lock.go

```diff
@@ -17,7 +17,7 @@ type repositoryLock struct {
 }
 
 // Lock acquires lock unless lock is already acquired with the same commit and allowConcurrent is set to true
-func (r *repositoryLock) Lock(path string, revision string, allowConcurrent bool, init func() error) (io.Closer, error) {
+func (r *repositoryLock) Lock(path string, revision string, allowConcurrent bool, init func() (io.Closer, error)) (io.Closer, error) {
```

Comment:
Should we have unit tests to verify that the closer produced by `directoryPermissionInitializer` does in fact set the correct permissions when a lock is acquired and when it is released?

Reply:
Done. Added tests